packages/roosterjs-editor-api/lib/utils/execCommand.ts

```diff
@@ -32,6 +38,17 @@ export default function execCommand(editor: IEditor, command: DocumentCommand) {
             });
         }
     } else {
-        editor.addUndoSnapshot(formatter, ChangeSource.Format);
+        editor.addUndoSnapshot(() => {
+            if (selection.type == SelectionRangeTypes.TableSelection) {

```

Comment:
Can we always use selection.ranges, no matter what selection type is?

Reply:
Sounds good.